File: day1/main_1_2.py
ans = [0, 0, 0]
line = input()
while len(line) > 0:
    # New elf
    tot = 0
    while len(line) > 0:
        tot += int(line)
        line = input()
    
    i = ans.index(min(ans))
    if ans[i] < tot:
        ans[i] = tot
    line = input()

print(sum(ans))

File: day1/main_1_2.rs
use std::fs;

fn main() {
    let mut ans = [0, 0, 0];
    let mut tot = 0;
    for i in fs::read_to_string("input").unwrap().lines(){

        if i.len() > 0 {
            tot += i.parse::<u32>().unwrap();
        }
        else{
            let (i, val) : (usize, &u32) = ans.iter().enumerate().min_by_key(|(_, &v)| v).unwrap();

            if val < &tot {
                ans[i] = tot;
            }
            tot = 0;
        }

    }

    println!("{}", ans.iter().sum::<u32>());
}